File: Chapter_4/project_4_5.py
barcode = int(input('Enter the 11 digit serial number: '))
barcode_list = [int(i) for i in str(barcode)]

even = sum(barcode_list[1::2])
odd = sum(barcode_list[::2])

check_digit = 9 - (((((even * 3) + odd) - 1) % 10))
print(f"Check digit {check_digit}")

File: Chapter_4/project_4_5.pl
use strict;
use warnings;
use List::Util qw( sum ); 

print 'Enter the 11 digiet serial number: ';
chomp( my @barcode = split //, <STDIN> );

#see https://stackoverflow.com/a/54436775/461887
my $sum_even_idxs = sum @barcode[0,2,4,6,8,10];
my $sum_odd_idxs  = sum @barcode[1,3,5,7,9];

my $check_digit = 9 - ( ( ( ( ( $sum_even_idxs * 3 ) + $sum_odd_idxs ) - 1 ) % 10 ) );
printf "Check digit $check_digit\n";

1;

